distributed/utils.py

```diff
@@ -777,17 +778,30 @@ def _maybe_complex(task):
 
 
 def convert(task, dsk, extra_values):
-    if type(task) is list:
+    typ = type(task)
+    if typ is tuple and len(task) > 1 and type(task[0]) is SubgraphCallable:
```

Comment:
`len(task) > 1` seems unnecessary.  I suspect that `type(task[0]) is ...` is about as fast and I wouldn't be surprised if, in the future, tasks without args here are possible.